### PR TITLE
update jackson and arrow version

### LINKED ISSUE #67: ["Request"] Upgrade Jackson to 2.13.1
What version are you currently using?
0.13.2

What would you like to see?
Jackson 2.13.1 is released, we would like to keep arrow integrations up to date

### PR DESCRIPTION
**Pull request by myuwono**
### In this PR:
- Update Kotlin to 1.6.10
- Update Jackson to 2.13.1
- Update arrow to 1.0.0
- Update test engine to Kotest 5.0.2
- Update gradle to 6.1.1

fixes #67

### CODE REVIEW

**Comment by myuwono** [comment]
@nomisRev i'm not sure who to tag for this repo. I wonder if we should publish the jackson modules separately vs the main arrow proper?

File: arrow-integrations-jackson-module/src/main/kotlin/arrow/integrations/jackson/module/NonEmptyListModule.kt [hunk 1]
```diff
@@ -16,7 +16,7 @@ import com.fasterxml.jackson.databind.type.CollectionType
 import com.fasterxml.jackson.databind.type.TypeFactory
 import com.fasterxml.jackson.databind.util.StdConverter
 
-object NonEmptyListModule : SimpleModule(PackageVersion.VERSION) {
+object NonEmptyListModule : SimpleModule(NonEmptyListModule::class.java.canonicalName, PackageVersion.VERSION) {
```
Comment (myuwono) [inline arrow-integrations-jackson-module/src/main/kotlin/arrow/integrations/jackson/module/NonEmptyListModule.kt:19]: This is now required due to the change in the way module registration detects uniqueness using module name which is done using a string. https://github.com/joansmith/jackson-databind/commit/d672b186bc549227afc210d481fa3dfde197b961

File: gradle.properties [hunk 2]
```diff
@@ -4,12 +4,11 @@ VERSION_NAME=1.0.0-SNAPSHOT
 LATEST_VERSION=0.13.2
 # Versions
 ANIMALS_SNIFFER_VERSION=1.5.0
-ARROW_VERSION=0.13.2
+ARROW_VERSION=1.0.0
```
Comment (myuwono) [inline gradle.properties:7]: @nomisRev i've bumped this from 0.13.2 to 1.0.0 - if i remember correctly they both should be backward compatible

File: gradle.properties [hunk 3]
```diff
@@ -4,12 +4,11 @@ VERSION_NAME=1.0.0-SNAPSHOT
...
 DOKKA_VERSION=0.10.0
-JACKSON_MODULE_KOTLIN_VERSION=2.12.3
-JUNIT_VINTAGE_VERSION=5.6.3
-KOTLIN_TEST_VERSION=3.3.3
-KOTLIN_VERSION=1.4.32
+JACKSON_MODULE_KOTLIN_VERSION=2.13.1
```
Comment (myuwono) [inline gradle.properties:9]: jackson is now updated to 2.13.1

File: gradle.properties [hunk 4]
```diff
@@ -4,12 +4,11 @@ VERSION_NAME=1.0.0-SNAPSHOT
...
+KOTEST_VERSION=5.0.2
+KOTLIN_VERSION=1.6.0
```
Comment (myuwono) [inline gradle.properties:11]: kotlin is updated to 1.6.0

File: gradle.properties [hunk 5]
```diff
@@ -4,12 +4,11 @@ VERSION_NAME=1.0.0-SNAPSHOT
...
 ANIMALS_SNIFFER_VERSION=1.5.0
-ARROW_VERSION=0.13.2
+ARROW_VERSION=1.0.0
 DOKKA_VERSION=0.10.0
-JACKSON_MODULE_KOTLIN_VERSION=2.12.3
-JUNIT_VINTAGE_VERSION=5.6.3
-KOTLIN_TEST_VERSION=3.3.3
-KOTLIN_VERSION=1.4.32
+JACKSON_MODULE_KOTLIN_VERSION=2.13.1
+KOTEST_VERSION=5.0.2
```
Comment (myuwono) [inline gradle.properties:10]: test engine is upgraded to kotest 5.0.2

**Review by i-walker — APPROVED** [review]
Thanks @myuwono 🙌🏾

File: gradle/libs.version.toml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+[versions]
+android = "7.0.4"
+arrow = "1.0.1"
+arrowGradleConfig = "0.6.0"
```
Comment (i-walker) [inline gradle/libs.version.toml:4]: this needs to be 0.9.1-alpha.3

File: build.gradle.kts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+plugins {
+  alias(libs.plugins.kotlin.jvm) apply false
+  alias(libs.plugins.arrowGradleConfig.kotlin)
+  alias(libs.plugins.kotest.multiplatform) apply false
+  alias(libs.plugins.kotlin.binaryCompatibilityValidator)
+  alias(libs.plugins.arrowGradleConfig.nexus)
+  alias(libs.plugins.arrowGradleConfig.versioning)
+  alias(libs.plugins.arrowGradleConfig.publish)
```
Comment (i-walker) [inline build.gradle.kts:8]:
```suggestion
  alias(libs.plugins.dokka)
  alias(libs.plugins.animalSniffer) apply false
  alias(libs.plugins.kotest.multiplatform) apply false
  alias(libs.plugins.kotlin.binaryCompatibilityValidator)
  alias(libs.plugins.arrowGradleConfig.nexus)
  alias(libs.plugins.arrowGradleConfig.versioning)
```

File: gradle/libs.version.toml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+javiersc-semver-semverGradlePlugin = { module = "com.javiersc.semver:semver-gradle-plugin", version.ref = "javierscSemverGradlePlugin" }
+kotlin-gradlePlugin = { module = "org.jetbrains.kotlin:kotlin-gradle-plugin", version.ref = "kotlin" }
+kotlin-scriptingJsr223Unshaded = { module = "org.jetbrains.kotlin:kotlin-scripting-jsr223-unshaded", version.ref = "kotlin" }
+spotless = { module = "com.diffplug.spotless:spotless-plugin-gradle", version.ref = "spotless" }
+
+[plugins]
+arrow-gradleConfig-formatter = { id = "io.arrow-kt.arrow-gradle-config-formatter", version.ref = "arrowGradleConfig" }
+arrow-gradleConfig-versioning = { id = "io.arrow-kt.arrow-gradle-config-versioning", version.ref = "arrowGradleConfig" }
+dokka = { id = "org.jetbrains.dokka", version.ref = "dokka" }
+kotlin-jvm = { id = "org.jetbrains.kotlin.jvm", version.ref = "kotlin" }
```
Comment (i-walker) [inline gradle/libs.version.toml:35]: you're missing kapt unless this is not needed anymore

Comment (myuwono) [reply gradle/libs.version.toml:35]: arrow proper is using ksp?

File: gradle/wrapper/gradle-wrapper.properties [hunk 9]
```diff
@@ -1,5 +1,5 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-6.0.1-all.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
```
[the remark this answers is not in the document]
Comment (myuwono) [reply gradle/wrapper/gradle-wrapper.properties:3]: awesome, I'll raise another issue to track the gradle upgrade

File: gradle.properties (hunk 2 above)
Comment (myuwono) [reply gradle.properties:7]: i think i hit another problem with the ank-gradle-plugin with 1.0.1 so i'm keeping it 1.0.0 for the time being, let's address it in the new issue

**Comment by i-walker** [comment]
Lets get this merged :D And do the other changes in another PR Thanks ✊🏾

**Pull request merged**